network/p2p/conduit/conduit.go

```diff
@@ -5,18 +5,33 @@ import (
 	"fmt"
 
 	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
 	"github.com/onflow/flow-go/network"
 )
 
 // DefaultConduitFactory is a wrapper around the network Adapter.
 // It directly passes the incoming messages to the corresponding methods of the
 // network Adapter.
 type DefaultConduitFactory struct {
+	component.Component
+	cm      *component.ComponentManager
 	adapter network.Adapter
 }
 
 func NewDefaultConduitFactory() *DefaultConduitFactory {
-	return &DefaultConduitFactory{}
+	d := &DefaultConduitFactory{}
+	cm := component.NewComponentManagerBuilder().
+		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
```

Comment:
Why do we need this worker?

Reply:
added clarification https://github.com/onflow/flow-go/pull/2308/commits/a42ae4221c601f2e6f44880991b5679a20788a5b